src/testresult.cpp

```diff
@@ -0,0 +1,57 @@
+#include <QDebug>
+
+#include "appsettings.h"
+#include "sfsecrethelper.h"
+#include "upload.h"
+#include "testresult.h"
+
+TestResult::TestResult(QObject *parent)
+    : QObject(parent)
```

Comment:
Your `TestResult::testClassLabel()` method is generating a SIGABRT at start up for me because `m_result` is out of range. You'll need to initialise it to avoid this.
```suggestion
    : QObject(parent)
    , m_result(Result::NoResult)
```

Reply:
Ok, thx. I believed enum classes autoinit with 0 :(